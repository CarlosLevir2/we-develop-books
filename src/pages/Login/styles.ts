import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  height: 100%;
  background-color: ${({ theme }) => theme.colors.background};
  padding: 2% 6%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const FormContainer = styled.div`
  width: 300px;
  height: 500px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const LoginInput = styled.input`
  padding: 1rem;
  border: ${({ theme }) => `2px solid ${theme.colors.accent.primary}`};
  max-width: 80%;
  width: 35rem;
  height: 3.8rem;
  border-radius: 0.4rem;
  margin-bottom: 0.5rem;
`;

export const LoginButton = styled.button`
  border: 0;
  max-width: 80%;
  width: 35rem;
  height: 3.8rem;
  background-color: ${({ theme }) => theme.colors.accent.primary};
  border-radius: 0.4rem;
  color: ${({ theme }) => theme.colors.text.highlight};
  cursor: pointer;
`;
