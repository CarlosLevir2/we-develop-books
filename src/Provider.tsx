import { QueryClientProvider } from 'react-query';
import { ThemeProvider } from 'styled-components';
import queryClient from './services/queryClient';
import { theme } from './styles/theme';

type Props = {
  children: React.ReactNode;
};

export default function Provider({ children }: Props) {
  return (
    <ThemeProvider theme={theme}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  );
}
