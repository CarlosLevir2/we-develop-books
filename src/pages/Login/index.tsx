import { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import * as S from './styles';

export default function Login() {
  const navigate = useNavigate();

  const handleSignIn = useCallback(() => {
    navigate('/home');
  }, [navigate]);

  return (
    <S.Container>
      <S.FormContainer>
        <S.LoginInput placeholder="Email" />
        <S.LoginInput placeholder="Password" />
        <S.LoginButton onClick={handleSignIn}>Login</S.LoginButton>
      </S.FormContainer>
    </S.Container>
  );
}
