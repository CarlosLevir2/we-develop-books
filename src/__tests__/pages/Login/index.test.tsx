import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Login from '../../../pages/Login';
import Provider from '../../../Provider';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

test('renders login page', () => {
  render(
    <Provider>
      <Login />
    </Provider>
  );
  const linkElement = screen.getByText(/Login/i);
  expect(linkElement).toBeInTheDocument();
});

test('match snapshot', () => {
  const { container } = render(
    <Provider>
      <Login />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});

test('push to home when login is clicked', async () => {
  render(
    <Provider>
      <Login />
    </Provider>
  );

  const linkElement = screen.getByText(/Login/i);
  userEvent.click(linkElement);

  expect(mockNavigate).toHaveBeenCalled();
});
