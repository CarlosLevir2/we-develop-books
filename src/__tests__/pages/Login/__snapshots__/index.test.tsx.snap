// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL ixnEgc"
  >
    <div
      class="sc-gsDKAQ iCRMHx"
    >
      <input
        class="sc-dkPtRN efTCSp"
        placeholder="Email"
      />
      <input
        class="sc-dkPtRN efTCSp"
        placeholder="Password"
      />
      <button
        class="sc-hKwDye zJlkb"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
